File: Day1/solution.js
// Find the two entries that sum to 2020; what do you get if you multiply them together?
const fs = require('fs');

const data = fs.readFileSync('./input.txt', 'utf-8').split('\r\n').map(Number);

// Part 1
for (let i = 0; i < data.length - 1; i++) {
    for (let j = i + 1; j < data.length; j++) {
        if (data[i] + data[j] === 2020) {
            console.log('Part 1: ', data[i] * data[j]);
            break;
        }
    }
}

// Part 2
for (let i = 0; i < data.length - 1; i++) {
    for (let j = i + 1; j < data.length; j++) {
        let remainder = 2020 - data[i] - data[j];
        if (data.indexOf(remainder) !== -1) {
            console.log('Part 2: ', data[i] * data[j] * remainder);
            break;
        }
    }
}

File: Day1/solution.py
data = [int(x) for x in open('./input.txt')]

# Part 1
for num in data:
    r = 2020 - num
    if r in data:
        print(num * r)
        break

# Part 2
for i in range(len(data)):
    for j in range(i+1, len(data)):
        remainder = 2020 - data[i] - data[j]
        if remainder in data:
            print(data[i] * data[j] * remainder)
            break
